File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int toDelete = s.find(part);
        while (toDelete != string::npos) {
            s.erase(toDelete, part.size());
            toDelete = s.find(part);
        }
        return s;
    }
};

File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.go
func removeOccurrences(s string, p string) string {
    for i, n := 0, len(p); i + n <= len(s); i++ {
        if s[i : i + n] == p {
            s = s[:i] + s[i + n :]
            i = -1
        }
    }
    
    return s
}